src/components/plots/barPlot.ts

```diff
@@ -38,6 +38,30 @@ export module Plots {
       this.attr("width", () => this._getBarPixelWidth());
     }
 
+    public x(): Plots.AccessorScaleBinding<X, number>;
+    public x(x: number | Accessor<number>): Bar<X, Y>;
+    public x(x: X | Accessor<X>, xScale: Scale<X, number>): Bar<X, Y>;
+    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: Scale<X, number>): any {
+      if (x == null) {
+        return super.x();
+      }
+      super.x(<X | Accessor<X>>x, xScale);
+      this._updateValueScale();
```

Comment:
For `barPlot`, the constructor is calling `this.baseline(0)`, which calls `this._updateValueScale`, but there is no scale, so this needs to be called here as a workaround. Also, we don't know if `x` or `y` is the value scale, so has to be called in 2 places